scripts/tuple_tree_generator/tuple_tree_generator/templates/struct_late.h.tpl

```diff
@@ -161,19 +163,36 @@ inline Upcastable/*= struct.name =*/ make/*= struct.name =*/(Args &&...A) {
   return Upcastable/*= struct.name =*/::make<T>(std::forward<Args>(A)...);
 }
 
+inline Upcastable/*= struct.name =*/
+copy/*= struct.name =*/(const /*= struct.name =*/ &From) {
+  Upcastable/*= struct.name =*/ Result;
+  upcast(&From, [&Result]<typename T>(const T &Upcasted) {
+    Result = Upcastable/*= struct.name =*/::make<T>(Upcasted);
+  });
+  return Result;
+}
```

Comment:
I feel like we can just make a `model::copy` template function for this, why generate?
At least factor implementations.

Reply:
Done!